File: leetcode/algorithms/605_can_place_flowers/main.dart
class CanPlaceFlowers {
  bool canPlaceFlowers(List<int> flowerbed, int n) {
    if (flowerbed.length == 1) {
      return flowerbed.first == 0 || n == 0;
    }
    for (int i = 0; i < flowerbed.length && n > 0; i++) {
      if (i == 0) {
        if (flowerbed[i] == 0 && flowerbed[i + 1] == 0) {
          flowerbed[i] = 1;
          n--;
        }
      } else if (i == flowerbed.length - 1) {
        if (flowerbed[i - 1] == 0 && flowerbed[i] == 0) {
          flowerbed[i] = 1;
          n--;
        }
      } else {
        if (flowerbed[i - 1] == 0 &&
            flowerbed[i] == 0 &&
            flowerbed[i + 1] == 0) {
          flowerbed[i] = 1;
          n--;
        }
      }
    }
    return n == 0;
  }

  // Solution
  bool solution(List<int> flowerbed, int n) {
    int len = flowerbed.length;
    for (int i = 0; i < len; i++) {
      bool left = i == 0 || flowerbed[i - 1] == 0;
      bool right = i == len - 1 || flowerbed[i + 1] == 0;
      if (left && right && flowerbed[i] == 0) {
        flowerbed[i] = 1;
        n--;
      }
    }
    return n <= 0;
  }
}

File: leetcode/algorithms/605_can_place_flowers/main.cpp
#include <vector>
using namespace std;

class CanPlaceFlowers {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        if (flowerbed.size() == 1) {
            return flowerbed.at(0) == 0 || n == 0;
        }
        for (int i = 0; i < flowerbed.size() && n > 0; i++) {
            if (i == 0) {
                if (flowerbed[i] == 0 && flowerbed[i + 1] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            } else if (i == flowerbed.size() - 1) {
                if (flowerbed[i - 1] == 0 && flowerbed[i] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            } else {
                if (flowerbed[i - 1] == 0 && 
                    flowerbed[i] == 0 &&
                    flowerbed[i + 1] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            }
        }
        return n == 0;
    }

    // Solution
    bool solution(vector<int>& flowerbed, int n) {
        if (n == 0) {
            return true;
        }
        for (int i = 0; i < flowerbed.size(); i++) {
            if (flowerbed[i] == 0 && (i == 0 || flowerbed[i - 1] == 0) &&
                (i == flowerbed.size() - 1 || flowerbed[i + 1] == 0)) {
                flowerbed[i] = 1;
                n--;
                if (n == 0) {
                    return true;
                }
            }
        }
        return false;
    }
};
